src/Digdir.Domain.Dialogporten.Infrastructure/HealthChecks/WarmupService.cs

```diff
@@ -0,0 +1,250 @@
+            {
+                IsSilentUpdate = true,
+                DialogId = result.AsT0.DialogId,
+            }, cancellationToken);
+        }
+    }
+
+    private static async Task PerformEndUserRequests(ISender sender, CancellationToken cancellationToken)
+    {
+        await sender.Send(GetEnduserSearchDialogQuery(), cancellationToken);
+    }
+
+    private static void CreateEndUserHttpContext(IServiceProvider scopedProvider)
+    {
+        CreateHttpContext(scopedProvider, new List<Claim>
+        {
+            new("scope", "digdir:dialogporten"),
+            new("pid", EndUserPid)
+        });
+    }
+
+    private static void CreateServiceOwnerHttpContext(IServiceProvider scopedProvider)
+    {
+        CreateHttpContext(scopedProvider, new List<Claim>
+        {
+            new("scope", "digdir:dialogporten.serviceprovider digdir:dialogporten.serviceprovider.admin"),
+            new("consumer", $"{{\"authority\":\"iso6523-actorid-upis\",\"ID\":\"0192:{ServiceOwnerOrgNo}\"}}")
+        });
+    }
+
+    private static void CreateHttpContext(IServiceProvider scopedProvider, List<Claim> claims)
+    {
+        var httpContextAccessor = scopedProvider.GetRequiredService<IHttpContextAccessor>();
+        var httpContext = new DefaultHttpContext
+        {
+            RequestServices = scopedProvider
+        };
+
+        var identity = new ClaimsIdentity(claims, "WarmupAuth");
+        httpContext.User = new ClaimsPrincipal(identity);
+        httpContextAccessor.HttpContext = httpContext;
+    }
+
+    private static async Task PerformLocalHttpRequests(CancellationToken cancellationToken)
+    {
+        var localKestrelAddress = GetLocalKestrelAdress();
+        if (localKestrelAddress is null)
+        {
+            return;
+        }
+
+        // Attempt to perform requests to both the WebAPI as well as the GraphQL endpoints
+        // to preload JWKs from issuer well-knowns and get som JIT-ing done
+        var assemblyName = Assembly.GetEntryAssembly()?.GetName().Name;
+        using var httpClient = new HttpClient();
+        httpClient.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer",
+            // Random invalid token
+            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlhdCI6MTUxNjIzOTAyMn0.KMUFsIDTnFmyG3nMiGM6H9FNFUROf3wh7SmqJp-QV30");
+        if (assemblyName == WebApiRuntimeAssembly)
+        {
+            var response = await httpClient.GetAsync(localKestrelAddress + $"api/v1/enduser/dialogs?Party={PartyId}", cancellationToken: cancellationToken);
+            Console.WriteLine(response.ReasonPhrase);
+        }
+        else if (assemblyName == GraphQlRuntimeAssembly)
+        {
+            // Perform GraphQL warmup requests here
+        }
+        // Don't bother with other runtimes (service, janitor)
+    }
+
+    private static Uri? GetLocalKestrelAdress()
```

Comment:
Hacky, men ASPNETCORE_URLS blir satt også for container apps og under lokal kjøring, så det burde kunne fungere.